cmd/snap/cmd_info.go

```diff
@@ -76,6 +77,69 @@ func init() {
 		}), nil)
 }
 
+func (iw *infoWriter) maybePrintHealth() {
+	if iw.localSnap == nil {
+		return
+	}
+	health := iw.localSnap.Health
+	if health == nil {
+		if iw.verbose {
+			fmt.Fprintln(iw, "health:\tunknown (no hook)")
+		}
+		return
+	}
+	if health.Status == "okay" && !iw.verbose {
+		return
+	}
+	buf := new(bytes.Buffer) // replace with a strings.Builder when >=1.10
+	fmt.Fprint(buf, health.Status)
+	if health.Message != "" {
+		fmt.Fprintf(buf, " (%s)", health.Message)
+	}
+	if health.Code != "" {
+		fmt.Fprintf(buf, " code %q", health.Code)
+	}
+	fmt.Fprintf(buf, ", last run %s for revision %s.\n", iw.fmtTime(health.Timestamp), health.Revision)
+
+	wrapFlow(iw, []rune(quotedIfNeeded(buf.String())), "health:\t", iw.termWidth)
```

Comment:
I don't remember agreeing that this would be a one liner, especially in the verbose case. In the notes I see:

```
health:
	status: error
	message: Something went wrong.
	checked: 4 minutes ago
	code: <code> # Optional
```